src/snackbar.tsx

```diff
@@ -128,9 +164,31 @@ const Snackbar: React.FC<Props> = ({
 
     React.useEffect(() => {
         if (renderNative) {
-            nativeMessage({message, duration, buttonText, type}).then(onCloseRef.current);
+            // these are the duration values understood by native app, other values will be ignored
+
+            nativeMessage({
+                message,
+                // @ts-expect-error duration can be 'PERSISTENT' in new webview-bridge lib versions, and old apps will ignore it
+                duration: duration === Infinity ? 'PERSISTENT' : undefined,
+                buttonText,
+                type,
+                ...{withDismiss},
+            }).then((result: unknown) => {
+                // there are terser ways to do this checks, but this one satisfies TS
```

Comment:
perhaps you could type the result like:
```
result: undefined | {action?: string}
```

Reply:
TS don't like this because current (old) version of webview-bridge returns void as result